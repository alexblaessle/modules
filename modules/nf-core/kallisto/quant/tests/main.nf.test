nextflow_process {

    name "Test Process KALLISTO_QUANT"
    script "../main.nf"
    process "KALLISTO_QUANT"
    tag "modules"
    tag "modules_nfcore"
    tag "kallisto"
    tag "kallisto/quant"

    setup {
            run("KALLISTO_INDEX") {
                script "../../index/main.nf"
                process {
                """
                input[0] = [
                    [ id:'test_fasta' ], // meta map
                    [ file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true) ]
                ]
                """
            }
            }
        }

    test("sarscov2 single-end") {
        config "./single_end.config"

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:true ],
                    [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
                ]
                input[1] = KALLISTO_INDEX.out.index
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
		{ assert snapshot(process.out.results.get(0).get(1)).match("abundance_tsv_single") },
                { assert snapshot(process.out.log).match("se_log") },
                { assert snapshot(process.out.versions).match("se_versions") }
            )
        }

    }
}
