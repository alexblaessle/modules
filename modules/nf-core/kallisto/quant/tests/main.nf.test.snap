{
    "se_versions": {
        "content": [
            [
                "versions.yml:md5,f981ad0cc089194a8fb00a47948eea94"
            ]
        ],
        "timestamp": "2023-10-30T22:11:17.982220232"
    },
    "se_log": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.log.txt:md5,9c166f0c50cd4fdbdbf1bff9d5d8aba2"
                ]
            ]
        ],
        "timestamp": "2023-10-30T22:11:17.973230763"
    },
    "abundance_tsv_single": {
        "content": [
            [
                "abundance.h5:md5,56f323d9ba10b4cc6ae3d28830e63efe",
                "abundance.tsv:md5,8a4afe91e6a75b4e619daaf664eb7d9b",
                "run_info.json:md5,aca46f70b74450a01cb60bc2d4073285"
            ]
        ],
        "timestamp": "2023-11-01T19:44:14.303819267"
    }
}